lib/iterable_basics.dart

```diff
@@ -116,8 +116,11 @@ extension IterableBasics<E> on Iterable<E> {
   /// ```dart
   /// ['a', 'aaa', 'aa'].maxBy((e) => e.length).value; // 'aaa'
   /// ```
-  E? maxBy(Comparable Function(E) sortKey) =>
-      this.max((a, b) => sortKey(a).compareTo(sortKey(b)));
+  E? maxBy(Comparable Function(E) sortKey) {
```

Comment:
If you could also do this with `sortBy`...

![8c587bf046fc7bd54f72bfc1476ab2e1](https://user-images.githubusercontent.com/4229329/168404932-e8a49608-94a6-4c76-bf7f-68b138244c63.jpg)

Reply:
Done. Since they're in two separate extension classes, I followed the same pattern as slice and put the shared logic under src/